torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -685,240 +685,103 @@ def __init__(
                 param_init_fn=param_init_fn,
                 device_id=device_id,
                 sync_module_states=sync_module_states,
+                forward_prefetch=forward_prefetch,
```

Comment:
unrelated to this PR, but it seems easy to forget to update the constructor kwargs in all the right places, and less likely to be caught in CR because there would be no code change. wonder if we can do anything about it

Reply:
I am not going to pursue this for now because for the real non-recursive wrapping, we do not initialize using this code path anymore, so this will be obsolete in the future.